### PR TITLE
Relax use-count check to allow for 0

### PR DESCRIPTION
**Pull request by soulitzer**
Previously we require tensor use count to be exactly 1. We should actually allow for use count to be zero as well. Use count is zero when an undefined tensor is returned, and this is common in backward functions that have multiple outputs.

In this PR I also remove some entries from the skip list that should be covered by this change: they return multiple tensors AND are backward functions. Batch norm is also known to return undefined tensors when `training=False`.

Related issue: https://github.com/pytorch/pytorch/issues/60426

### CODE REVIEW

File: tools/autograd/gen_variable_type.py [hunk 1]
```diff
@@ -180,8 +179,8 @@
 if (${tensor_name}_impl_saved) AT_ASSERT(${tensor_name}_impl_saved == ${tensor_name}.getIntrusivePtr());
 """)
 
-ENFORCE_TENSOR_IMPL_USE_COUNT_EQUALS_ONE = CodeTemplate("""\
-AT_ASSERT(${tensor_name}.use_count() == 1, "function: ${fn_name}");
+ENFORCE_TENSOR_IMPL_USE_COUNT_GT_OR_EQ_ONE = CodeTemplate("""\
```
Comment (albanD) [inline tools/autograd/gen_variable_type.py:182]: The name should be LT_ONE no?

**Review by albanD — APPROVED** [review]
LGTM
Just variable name to update.

File: tools/autograd/gen_variable_type.py (hunk 1 above)
Comment (soulitzer) [reply tools/autograd/gen_variable_type.py:182]: Whoops
